client/object_get.go

```diff
@@ -448,7 +448,12 @@ func (c *Client) ObjectHead(ctx context.Context, containerID cid.ID, objectID oi
 		res.hdr = v
 	}
 
-	return &res, nil
+	var hdr object.Object
+	if !res.ReadHeader(&hdr) {
+		return nil, errors.New("empty header")
```

Comment:
Can't there be some protocol-level error there? @carpawell?

Reply:
@roman-khimov, you mean that we can return some "bad response" error here? or you mean that header is missing cause we have not checked it?

1. there is no such err, but we can add smth like https://github.com/nspcc-dev/neofs-api/issues/269 but for client side
2. no, we checked statuses on `processResponse` call and no error retuned so we are looking for a header

Reply:
Can there be some error returned from the server? Like we're trying to read from a stream, but server closes it for whatever reason.

Reply:
can't say i understand you. `ReadHeader` does not do any RPC, just struct conversions. all the network work has been done (and checked) above

Reply:
Ah, I see now. Then we don't need `resObjectHead` at all.

Reply:
> don't need resObjectHead at all

yeap, that is why i requested dropping its publicity. but complete dropping is even sweeter